File: frontend/src/types/generate.ts
import type { Letter } from './letter';

export type Tone = 'Friendly' | 'Formal' | 'Emotional' | 'Encouraging';

export const TONES: Tone[] = ['Friendly', 'Formal', 'Emotional', 'Encouraging'];

export type Gender = 'male' | 'female';

export const GENDERS: Gender[] = ['male', 'female'];

export interface GenerateLetterRequest {
  title: string;
  category: string;
  description: string;
  tone: Tone;
  gender: Gender;
}

export interface GenerateLetterResponse {
  letter: Letter;
  saved: boolean;
}

export interface LetterFormData {
  title: string;
  category: string;
  description: string;
  tone: Tone;
  gender: Gender;
}

export interface LetterFormErrors {
  title?: string;
  category?: string;
  description?: string;
  tone?: string;
  gender?: string;
}

File: backend/app/schemas/generate.py
from enum import Enum

from pydantic import BaseModel, Field


class Tone(str, Enum):
    FRIENDLY = "Friendly"
    FORMAL = "Formal"
    EMOTIONAL = "Emotional"
    ENCOURAGING = "Encouraging"


class GenerateLetterRequest(BaseModel):
    title: str = Field(..., min_length=2, max_length=120)
    category: str = Field(..., min_length=2, max_length=60)
    description: str = Field(..., min_length=10, max_length=1000)
    tone: Tone


class GenerateLetterResponse(BaseModel):
    letter: "Letter"
    saved: bool = True


from app.schemas.letter import Letter  # noqa: E402

GenerateLetterResponse.model_rebuild()
